### ISSUE TITLE
ROCM 6.0 GA rocm_smi.h fails to compile, error: unknown type name ‘metrics_table_header_t’

### ISSUE THREAD

**Issue by ozziemoreno** [RadeonOpenCompute/rocm_smi_lib#147]
Function call `rsmi_dev_metrics_header_info_get` does not define `metrics_table_header_t* header_value` as struct

rsmi_dev_metrics_header_info_get(uint32_t dv_ind, metrics_table_header_t* header_value);

~$ gcc -I/opt/rocm/include test.c
```
In file included from test.c:1:
/opt/rocm/include/rocm_smi/rocm_smi.h:5370:51: error: unknown type name ‘metrics_table_header_t’
 5370 | rsmi_dev_metrics_header_info_get(uint32_t dv_ind, metrics_table_header_t* header_value);
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~
```
~$ cat test.c
```
#include <rocm_smi/rocm_smi.h> 
```
~$ dpkg -l | grep rocm
```
ii  rocm-core6.0.0                        6.0.0.60000-91~22.04                    amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-smi-lib6.0.0                     6.0.0.60000-91~22.04                    amd64        AMD System Management libraries
```

**Comment by dmitrii-galantsev** [comment]
We have a patch in the works by @oliveiradan that fixes it (and other metrics related issues).
Will keep it open until it's merged develop branch.

**Comment by bertwesarg** [comment]
Still broken with `6.0.1-108`.

**Comment by bertwesarg** [comment]
I can confirm, that this is solved with `6.0.1-112`. Thanks.

**Comment by dmitrii-galantsev** [comment]
Fixed in `6.0.1-112`. Thanks for confirming @bertwesarg !
